### PR TITLE
Remove max_input_len and max_num_sequences

### PR DESCRIPTION
**Pull request by yelite**
This PR removes `max_input_len` and `max_num_sequences`. User should specify `max_num_batched_tokens` instead.

cc @sunggg @masahi

### CODE REVIEW

File: serve/mlc_serve/model/tvm_model.py [hunk 1]
```diff
@@ -446,7 +446,7 @@ def init_tvm_model(
         LOG.info("Running memory profiling.")
         num_blocks = get_num_cache_blocks(
             model,
-            [engine_config.max_input_len] * engine_config.max_num_sequences,
+            [1] * engine_config.max_num_batched_tokens,
```
Comment (yelite) [inline serve/mlc_serve/model/tvm_model.py:449]: This change will make memory profiling overestimate memory usage and allocate smaller KV cache. I tested with mistral 7B on single H100, the number of blocks reduces from 28638 to 28389, which seems to be okay.

**Comment by masahi** [comment]
What's the intention of this change? (I remember @sunggg and I discussed making this change but forgot the reason).

Will ollm also explicitly configure `max_num_batched_tokens`?

File: serve/mlc_serve/model/tvm_model.py (hunk 1 above)
Comment (masahi) [reply serve/mlc_serve/model/tvm_model.py:449]: I remember doing this change and wondering why the profile result becomes different. Do you know why?

Comment (yelite) [reply serve/mlc_serve/model/tvm_model.py:449]: There will be more rows in the output logits (https://github.com/octoml/mlc-llm/blob/66a2e534ab4f6981acb0be117b99ca79ea862578/mlc_llm/relax_model/llama_batched_vllm.py#L481-L496). This seems to be the only place affected by the `seq_lens`

**Comment by yelite** [comment]
> What's the intention of this change? (I remember @sunggg and I discussed making this change but forgot the reason).

I think it's just to cleanup engine parameters that are unused and confusing. 

> Will ollm also explicitly configure `max_num_batched_tokens`?

Yes it should explicitly configure it unless it wants the default value (which should be good for model with short context window)

**Review by masahi — APPROVED** [review]

**Comment by sunggg** [comment]
Initially I thought having the separate `max_input_len` and `max_num_sequences` would be easier to understand, but realized `max_num_sequences` gives the wrong impression about the batch size. (got lots of questions about this as well.) I think the direct configuration of `max_num_batched_tokens` is more straightforward as this is what engine would use anyways.

File: serve/mlc_serve/model/tvm_model.py (hunk 1 above)
Comment (elvin-n) [reply serve/mlc_serve/model/tvm_model.py:449]: in the speculative decoding or logits workflow, it can be important and might lead to OOM issue

Comment (yelite) [reply serve/mlc_serve/model/tvm_model.py:449]: > in the speculative decoding or logits workflow, it can be important and might lead to OOM issue

@elvin-n Can you elaborate how OOM could occur? I believe this change makes memory profiling over-estimate memory consumption, thus allocating less KV cache blocks. In the worst case, the profiling is so conservative that it allocates not enough blocks for the max_context_length or even OOM during profiling. But it shouldn't OOM after server startup.
